### PR TITLE
Dead bodies cool to room temp

### LINKED ISSUE #78508: Bodies never cool down their body temperature naturally
## Reproduction:

image

This poor fucker's body temperatuer was so fucking high it literally outpaced my synthflesh patch factory's output. It went from roughly 233 from my patching to the 600 you see above in teh time it took me as a chemist to get to a surgyery room. 

The issue, and Ive seen this before, is that corpse body temp never goes down for some reason. Thsi sucks and makes treating husks very annoying

### PR DESCRIPTION
**Pull request by yorii**
## About The Pull Request

Fixes #78508

## Why It's Good For The Game

There was already code for heating dead bodies up with room temperature, now there is also code to cool dead bodies to room temperature.

## Changelog
:cl:
fix: dead bodies now cool down to room temperature over time
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/life.dm [hunk 1]
```diff
@@ -543,6 +543,9 @@
 	if(stat != DEAD) // If you are dead your body does not stabilize naturally
 		natural_bodytemperature_stabilization(environment, seconds_per_tick, times_fired)
 
+	else if(!on_fire && areatemp < bodytemperature) // lowers your dead body temperature to room temperature over time
+		adjust_bodytemperature((areatemp - bodytemperature), use_insulation=FALSE, use_steps=TRUE)
+
 	if(!on_fire || areatemp > bodytemperature) // If we are not on fire or the area is hotter
```
Comment (yorii) [inline code/modules/mob/living/carbon/life.dm:549]: Should this really have !on_fire?

**Comment by yorii** [comment]
I'm not sure if this current commit is preferable over the previous commit...?

**Comment by yorii** [comment]
Changed it back to the first edit, there were some side effects from just altering the second if-statement.

**Review by Kapu1178 — CHANGES_REQUESTED** [review]
Humans never call the carbon level handle_environment() proc. Something is wrong here.

**Comment by ABS-Helmet** [comment]
Do they warm up the room a bit too?

**Comment by yorii** [comment]
> Do they warm up the room a bit too?

Haha no I don't think there's any such mechanic, other than a body being set on fire :p

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**